packages/core/src/common/keybinding.ts

```diff
@@ -311,37 +338,66 @@ export class KeybindingRegistry {
+        }
 
-        for (const binding of bindings) {
-            const context = binding.context
-                ? this.contextRegistry.getContext(binding.context)
-                : undefined;
+        if (this.timeoutTimer) {
+            clearTimeout(this.timeoutTimer);
+        }
 
-            /* Only execute if it has no context (global context) or if we're in
-               that context.  */
-            if (!context || context.isEnabled(binding)) {
+        this.keySequence.push(keyCode);
+        const bindings = this.getKeybindingsForKeySequence(this.keySequence);
 
-                if (this.isPseudoCommand(binding.command)) {
-                    /* Don't do anything, let the event propagate.  */
-                } else {
-                    const command = this.commandRegistry.getCommand(binding.command);
-                    if (command) {
-                        const commandHandler = this.commandRegistry.getActiveHandler(command.id);
+        if (bindings.full.length > 0) {
+            this.keySequence = [];
+
+            for (const binding of bindings.full) {
+                const context = binding.context
+                    ? this.contextRegistry.getContext(binding.context)
+                    : undefined;
+
+                /* Only execute if it has no context (global context) or if we're in
+                   that context.  */
+                if (!context || context.isEnabled(binding)) {
+
+                    if (this.isPseudoCommand(binding.command)) {
+                        /* Don't do anything, let the event propagate.  */
+                    } else {
+                        const command = this.commandRegistry.getCommand(binding.command);
+                        if (command) {
+                            const commandHandler = this.commandRegistry.getActiveHandler(command.id);
 
-                        if (commandHandler) {
-                            commandHandler.execute();
+                            if (commandHandler) {
+                                commandHandler.execute();
+                            }
+
+                            /* Note that if a keybinding is in context but the command is
+                               not active we still stop the processing here.  */
+                            event.preventDefault();
+                            event.stopPropagation();
                         }
+                    }
 
-                        /* Note that if a keybinding is in context but the command is
-                           not active we still stop the processing here.  */
-                        event.preventDefault();
-                        event.stopPropagation();
+                    break;
+                } else {
+                    /* Try to match a partial binding.  */
+                    if (bindings.partial.length > 0) {
+                        this.keySequence.push(keyCode);
+                        this.timeoutTimer = setTimeout(() => {
```

Comment:
Do we need a timeout? I see that VS code notifies a user about the current sequence in the status bar and that it is waiting for next character. If nothing is matched eventually it tells that is no such command for such key combination.

Reply:
Well the timeout is the would  to do like vscode and  `eventually tell that there is no such command` the eventually is the timeout.

I think it's good to modify the status bar but I would make that in another PR as well as having a preference for the timeout as the preference-api is merged in core.

Reply:
ok, should we have another issue for it?

Reply:
Yes I'll open it with the others about the limitations once this is merged.